web_server/posts/models.py

```diff
@@ -36,19 +45,18 @@ class Post(models.Model):
     author = models.ForeignKey(Author, on_delete=models.PROTECT)
 
     # Storing a list of comma delineated categories
-    # @todo Should this be replaced with a many-to-many combo table and a separate 'categories' model?
-    # categories = models.CharField(max_length=256)
+    categories = models.ManyToManyField(Category, blank=True)
 
     # Count of comments
+    # @todo is this field strictly necessary? It should be computed not stored
     count = models.PositiveIntegerField()
 
     # Page Size
     # @todo What is a 'page'? A page of comments?
     size = models.PositiveIntegerField()
 
     # Comments
-    # @todo obviously this should not be implemented in this model, it should be implemented as
-    # a foreign key in the comments model
+    # implemented as foreign key in comments model
```

Comment:
I believe that comments should be the many to many model in this class:
comments = models.ManyToManyField(Comment)

Then getting the count is as simple as:
postObj.comments.all().count()

Reply:
The solution for counting is just as simple if we use a foreign key: https://stackoverflow.com/questions/32998591/django-count-of-foreign-key-model
Additionally foreign keys model the many-to-one relationship model that comments would have to posts correctly.

Reply:
Sounds good!